site/content/contribute/server/feature-flags.md

```diff
@@ -156,19 +157,17 @@ When the feature is rolled out to customers, logs will show if there are crashes
 
 ## Self-hosted releases
 
-For a feature-flagged feature to be included in a self-hosted release, the feature flag should be removed.
-
-Feature flags are generally off by default and self-hosted releases do not contact the management system. Therefore feature flags that are not ready for a self-hosted release will be automatically disabled for all self-hosted releases.
+For a feature-flagged feature to be included in a self-hosted release, the feature flag should be removed. Feature flags are generally off by default and self-hosted releases do not contact the management system. Therefore feature flags that are not ready for a self-hosted release will be automatically disabled for all self-hosted releases.
```

Comment:
What message are you trying to convey here?

Reply:
I've clarified the text further now.

Reply:
> For a feature-flagged feature to be included in a self-hosted release, the feature flag should be removed.

For Insights we wanted the feature to be enabled by default on-prem, but we also wanted the feature flag left in the code to be available (as an escape hatch) to disable Insights in case things went terribly wrong.

According to this description that wouldn't be possible.

Reply:
@mkraft I clarified the text a bit in the latest commit. You can find the updated text [here](https://github.com/mattermost/mattermost-developer-documentation/blob/e7cd803a3c89c15c4b93181bd0f566cc0c68d2a1/site/content/contribute/server/feature-flags.md#self-hosted-releases).